3drepo/3drepo.io
# Issue 5235 - Custom tickets show only allowed users/jobs in the assignee list

This fixes #5235 

#### Description
Each model is linked to some users and jobs. To be linked to a user, it means that that user has at least viewer access. To be linked to job, it means at least of one of the linked user is assigned to that job.

Given this, only linked jobs and users are considered valid for a job/user list. Note, for the `assignee property` (i.e. the one in the ticket top panel), only user with a role higher than viewer are considered valid.

As it is possible to have some users/jobs already assigned to user/job property, the invalid values will be indicated:
- a value that still exists in the teamspace will have a red background (for instance, the user has been demoted to `unassigned` for that model);
- a value that cannot be found in the teamspace will be in different list section ("job or user not found"), with a _ad hoc_ circle popover, and with a red background

If it is a multi-select list, toggling ANY job/user, will remove all the invalid jobs/users from the list

#### Test cases
- Add some user/job to the list, and then unlink them. The user/job data should be available in the popover, but the background should be red.
- Add some user/job to the list and then remove them from the teamspace. The user/job data should be unavailable, the background should be red, and hovering the popover circle should better inform the user. Also, the latter users/jobs should be listed in a separate list section
